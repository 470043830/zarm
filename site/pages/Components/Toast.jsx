import Markdown from '@site/components/Markdown';

export default class Page extends Markdown {
  document = () => {
    return {
      document: require('@/components/toast/demo.md'),
      className: 'toast-page',
    };
  }
}
