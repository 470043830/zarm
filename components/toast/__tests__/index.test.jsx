import React from 'react';
import { render, shallow, mount, spyOn } from 'enzyme';
import toJson from 'enzyme-to-json';
import Toast from '../index';

describe('Toast', () => {
  const props = {
    mask: true,
    stayTime: 1,
    onMaskClick: jest.fn(),
  };
  it('renders correctly', () => {
    const wrapper = render(
      <Toast {...props}>foo</Toast>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('visible change false', () => {
    const wrapper = mount(
      <Toast visible>foo</Toast>,
    );
    wrapper.setProps({ visible: false });
  });

  it('stayTime', () => {
    jest.useFakeTimers();
    props.onClose = jest.fn();
    const wrapper = mount(
      <Toast stayTime={5000} visible {...props}>foo</Toast>,
    );
    jest.runAllTimers();
    wrapper.unmount();
  });

  it('stayTime is 0', () => {
    const wrapper = shallow(
      <Toast stayTime={0}>foo</Toast>,
    );
    wrapper.setProps({ visible: true });
  });

  it('static function show', () => {
    const wrapper = Toast.show();
    expect(wrapper).toMatchSnapshot();
  });

  it('static function hide', () => {
    const wrapper = Toast.hide();
    expect(wrapper).toMatchSnapshot();
  });
});
