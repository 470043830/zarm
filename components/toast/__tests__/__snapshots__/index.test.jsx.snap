// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders correctly 1`] = `
<div
  class="za-toast"
>
  <div
    class="za-toast__container"
  >
    foo
  </div>
</div>
`;
